core/templates/dev/head/pages/exploration_player/ConversationSkinDirective.js

```diff
@@ -714,14 +695,17 @@ oppia.directive('conversationSkin', [
 
             var timeAtServerCall = new Date().getTime();
             PlayerPositionService.recordAnswerSubmission();
-
-            $scope.answerIsCorrect = ExplorationEngineService.submitAnswer(
+            var currentEngineService =
+              ExplorationPlayerStateService.getCurrentEngineService();
+            $scope.answerIsCorrect = currentEngineService.submitAnswer(
               answer, interactionRulesService, function(
                   newStateName, refreshInteraction, feedbackHtml,
                   feedbackAudioTranslations, contentHtml, newParams,
                   refresherExplorationId, missingPrerequisiteSkillId,
-                  remainOnCurrentCard, wasOldStateInitial, isFirstHit) {
-                if (!_editorPreviewMode) {
+                  remainOnCurrentCard, wasOldStateInitial, isFirstHit,
+                  isFinalQuestion) {
```

Comment:
I thought we talked about getting rid of isFinalQuestion. This is not something that the conversation skin service needs to know.

Reply:
Created an [issue](https://github.com/oppia/oppia/issues/5516) for this, as discussed. After that refactor, this would also be fixed.